rules/numeric-separators-style.js

```diff
@@ -0,0 +1,142 @@
+'use strict';
+const merge = require('lodash/merge');
+const getDocumentationUrl = require('./utils/get-documentation-url');
+
+function addSeparator(value, {minimumDigits, groupLength}, fromLeft) {
+	const {length} = value;
+
+	if (length < minimumDigits) {
+		return value;
+	}
+
+	const parts = [];
+	if (fromLeft) {
+		for (let start = 0; start < length; start += groupLength) {
+			const end = Math.min(start + groupLength, length);
+			parts.push(value.slice(start, end));
+		}
+	} else {
+		for (let end = length; end > 0; end -= groupLength) {
+			const start = Math.max(end - groupLength, 0);
+			parts.unshift(value.slice(start, end));
+		}
+	}
+
+	return parts.join('_');
+}
+
+function addSeparatorFromLeft(value, options) {
+	return addSeparator(value, options, true);
+}
+
+function formatNumber(value, options) {
+	const parts = value.split('.');
+	const [decimal, fractional] = parts;
+
+	let formatted = addSeparator(decimal, options);
+	if (parts.length === 2) {
+		formatted += '.';
+		formatted += addSeparatorFromLeft(fractional, options);
+	}
+
+	return formatted;
+}
+
+function format(value, options) {
+	const {
+		prefix = '',
+		data
+	} = value.match(/^(?<prefix>0[box])?(?<data>.*)$/i).groups;
+
+	const formatOption = options[prefix.toLowerCase()];
+
+	if (prefix) {
+		return prefix + addSeparator(data, formatOption);
+	}
+
+	const {
+		number,
+		mark = '',
+		sign = '',
+		power = ''
+	} = value.match(/^(?<number>.*?)(?:(?<mark>e)(?<sign>[+-]?)?(?<power>\d+))?$/i).groups;
+
+	return formatNumber(number, formatOption) + mark + sign + addSeparator(power, options['']);
+}
+
+const create = context => {
+	const defaultOptions = {
+		hexadecimal: {minimumDigits: 0, groupLength: 2},
+		binary: {minimumDigits: 0, groupLength: 4},
+		octal: {minimumDigits: 0, groupLength: 4},
+		number: {minimumDigits: 5, groupLength: 3}
+	};
```

Comment:
Move defaultOptions to outer scope.

Reply:
Well.. As weird as it seems, I can't move it out of its scope... Then it is undefined in `create()`. Is it really a problem ?